BaseClasses.py

```diff
@@ -1323,7 +1323,7 @@ def advancement(self) -> bool:
     @property
     def is_event(self) -> bool:
         """Returns True if the address of this location is None, denoting it is an Event Location."""
-        return self.address is None
+        return type(self.address) is not int
```

Comment:
I'm wondering about the value of forbidding/allowing `IntEnum` for (non-event) locations,
since that's a difference that `type(...)` vs `isinstance` would make.

I can see some good reasons for wanting to use `IntEnum` (but not enough to motivate me to use it).

Reply:
As written, this code will always return true.  Is is a reference comparison (like === in javascript) not a type check.  It works with None because None is a singleton.
```
>>> 3 is int
<python-input-1>:1: SyntaxWarning: "is" with 'int' literal. Did you mean "=="?
  3 is int
False
```

This needs to use `isinstance()`

Reply:
Althouth there doesn't seem to be explicit style guidance for `==` vs `is` in this case (I personally prefer `==`), the code is functional since there is only a single instance of the `int` class. `type(3) is int` (for the value's type instead of just itself) is true.
Whether it is actually desirable to restrict to just the int class or allow subclasses is touched on above.